src/ferry_planner/schedule.py

```diff
@@ -254,36 +308,180 @@ def _refresh_task(self) -> None:
                 datetime.strptime(
                     td3,
-                    "%Hh %Mm",
+                    td3format,
                 ),
             ).total_seconds(),
         )
         sailing = FerrySailing(
             departure=departure,
             arrival=arrival,
             duration=duration,
+            notes=notes,
         )
-        yield sailing
+        sailings.append(sailing)
+    return sailings
+
+
+def parse_sailig_comment(comment: str) -> list[str]:
+    notes: list[str] = []
+    comment = comment.strip()
+    notes.append(comment)
+    pos = comment.find("Note:")
+    if pos > 0:
+        notes.append(comment[pos:])
+        comment = comment[:pos].strip()
+    if comment.startswith("Last "):
+        notes.append(comment)
+        comment = ""
+    notes[0] = comment  # replace original with truncated
+    return notes
+
+
+def get_seasonal_schedule_rows(soup: BeautifulSoup, date: datetime) -> Sequence[Tag] | None:
+    rows: Sequence[Tag] = []
+    form = soup.find("form", id="seasonalSchedulesForm")
+    if not isinstance(form, Tag):
+        return None
+    weekday_names = ("monday", "tuesday", "wednesday", "thursday", "friday", "saturday", "sunday")
+    weekday = weekday_names[date.weekday()]
+    for thead in form.find_all("thead"):
+        if thead.text.lower().strip().startswith(weekday):
+            rows = [x for x in itertools.takewhile(lambda t: t.name != "thead", thead.next_siblings) if x.name == "tr"]
+            break
+    return rows
+
+
+def get_seasonal_schedule_daterange_index(hrefs: Sequence[str], date: datetime) -> int:
+    for i, href in enumerate(hrefs):
+        dates = get_seasonal_schedule_daterange_from_url(href)
+        if dates and date.date() >= dates[0].date() and date.date() <= dates[1].date():
+            return i
+    return -1
+
+
+def get_seasonal_schedule_daterange_from_url(href: str) -> tuple[datetime, datetime] | None:
+    dates = href.replace("=", "-").replace("_", "-").split("-")[-2:]
+    expected_dates_count = 2
+    if (len(dates)) != expected_dates_count:
+        return None
+    date_from = datetime.strptime(dates[0], "%Y%m%d")
+    date_to = datetime.strptime(dates[1], "%Y%m%d")
+    return (date_from, date_to)
+
+
+def is_schedule_excluded_on_date(schedule_comment: str, date: datetime) -> bool:
+    if not schedule_comment:
+        return False
+    schedule_comment = schedule_comment.strip()
+    if schedule_comment.upper().startswith("ONLY"):
+        return not match_specific_schedule_date(schedule_comment, date)
+    if schedule_comment.upper().startswith(("EXCEPT", "NOT AVAILABLE")):
+        return match_specific_schedule_date(schedule_comment, date)
+    print("Unknown comment: " + schedule_comment)
+    return False
+
+
+def match_specific_schedule_date(schedule_dates: str, date: datetime) -> bool:
+    months: Sequence[str] = ["JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"]
```

Comment:
Can be moved to a module-level constant tuple

Reply:
at module level should it be uppercase or lowercase?

Reply:
The convention for constants is uppercase like `MONTH_NAMES`